futures-util/src/stream/try_stream/mod.rs

```diff
@@ -287,6 +302,27 @@ pub trait TryStreamExt: TryStream {
         assert_stream::<Result<Self::Ok, Fut::Error>, _>(OrElse::new(self, f))
     }
 
+    /// A future that completes after the given stream has been fully processed
+    /// into the sink and the sink has been flushed and closed.
+    ///
+    /// This future will drive the stream to keep producing items until it is
+    /// exhausted, sending each item to the sink. It will complete once the
+    /// stream is exhausted, the sink has received and flushed all items, and
+    /// the sink is closed. Note that neither the original stream nor provided
+    /// sink will be output by this future. Pass the sink by `Pin<&mut S>`
+    /// (for example, via `try_forward(&mut sink)` inside an `async` fn/block) in
+    /// order to preserve access to the `Sink`. If the stream produces an error,
+    /// that error will be returned by this future without flushing/closing the sink.
+    #[cfg(feature = "sink")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "sink")))]
+    fn try_forward<S>(self, sink: S) -> TryForward<Self, S>
+    where
+        S: Sink<Self::Ok, Error = Self::Error>,
+        Self: Sized,
+    {
+        assert_future::<Result<(), S::Error>, _>(TryForward::new(self, sink))
```

Comment:
```suggestion
        Self: Sized + Stream<Item = Result<Self::Ok, Self::Error>>,
    {
        assert_future::<Result<(), Self::Error>, _>(TryForward::new(self, sink))
```

You'll also need to `use crate::stream::Stream` at the top & fix a rebase conflict.